base/strings/io.jl

```diff
@@ -373,7 +373,10 @@ function unescape_string(io, s::AbstractString)
     end
 end
 
-macro b_str(s); :(Vector{UInt8}($(unescape_string(s)))); end
+macro b_str(s)
+    v = unsafe_wrap(Vector{UInt8}, unescape_string(s))
```

Comment:
This isn't performance sensitive to run the macro, but this would probably generate better runtime performance if it was making an actual copy here.